plugins/optimization-detective/detect.js

```diff
@@ -511,6 +518,26 @@ export default async function detect( {
 
 	// Stop observing.
 	disconnectIntersectionObserver();
+
+	const inpData = [];
+
+	onINP(
+		/**
+		 *
+		 * @param {INPMetric|INPMetricWithAttribution} metric
+		 */
+		( metric ) => {
+			if ( 'attribution' in metric ) {
+				// TODO: Store xpath instead?
+				inpData.push( {
+					value: metric.value,
+					rating: metric.rating,
+					interactionTarget: metric.attribution.interactionTarget,
+				} );
+			}
+		}
+	);
+
```

Comment:
> What's missing is passing the INP data to OD extensions.

Sp you'll be moving this code to https://github.com/swissspidy/od-debug-helper right? It can be registered as part of a new extension script module registered via the `od_extension_module_urls` filter. Then the `inpData` schema can be registered via the `od_url_metric_schema_root_additional_properties` filter.

```suggestion
```

Reply:
Yes. I have moved the PHP code already last night, but not the JS yet.